### DISCUSSION TITLE
Front Port CSV Upload Failure

### DISCUSSION THREAD

**Discussion by BH-NetAdmin** [netbox-community/netbox#8363] [Q&A]
Hi all,

I am getting this error (bellow) when trying to import front ports, how ever the rear ports are created. Any help is appreciated

<class 'dcim.models.device_components.FrontPort.rear_port.RelatedObjectDoesNotExist'>

FrontPort has no rear_port.

Python version: 3.8.10
NetBox version: 3.1.5

**Comment by candlerb** [comment]
You need to associate the front port with a rear port.  That is, you must give the name of the rearport that the frontport relates to.  This goes in the columns "rear_port" and "rear_port_position"

e.g. if your device has a single rear port called `R` which has 4 positions, and you want to create four ports `F[1-4]` linked to that rear port, it might be:

```
device,name,type,rear_port,rear_port_position
foo,F1,8P8C,R,1
foo,F2,8P8C,R,2
foo,F3,8P8C,R,3
foo,F4,8P8C,R,4
```

However if each rear port has only 1 position, meaning there is a 1:1 relation between rear port, then you use position "1" of each rearport.  Say that you have four rear ports `R[1-4]`, then to import the corresponding front ports `F[1-4]` would be:

```
device,name,type,rear_port,rear_port_position
foo,F1,8P8C,R1,1
foo,F2,8P8C,R2,1
foo,F3,8P8C,R3,1
foo,F4,8P8C,R4,1
```

**Reply by BH-NetAdmin** [reply]
Thank you for your response! it is wired it worked once after waiting 24 hours, but fails when i use the same template to add in more front ports.
This is the sheet I am working off [https://docs.google.com/spreadsheets/d/1y4NIPdIXHkfuQcF_9Z4c6wTOXGzfkADJILIlEGchCrI/edit?usp=sharing](url)

I think it is set up correctly

**Reply by candlerb** [reply]
First you have to import the rearports, then you have to import the frontports (you can't import a frontport until its corresponding rearport exists).  Can you show samples of the CSVs you're using to do that?

**Reply by BH-NetAdmin** [reply]
Here is the sheet template I used to import all rear ports [https://docs.google.com/spreadsheets/d/1ZlXOpBPCQKhuYtlvmT98fXLuEoq3-DBiieiz7yUH95w/edit?usp=sharing](url). All rear ports have been imported based on that sheet. As the ui is showing the rear ports existing, is there possibly a back end issue?

**Reply by candlerb** [reply]
(Aside: your markdown is broken.  It should be: `[url](http://example.com/)` or simply `<http://example.com>`)

In the two spreadsheets you've shown, the device names are different.  You created rearports on device `HS-IDF2 PP 01` and `HS-IDF2 PP 02`, but frontports on devices `CH-IDF2 PP 1`, `CH-IDF2 PP 2`, `CH-IDF3 PP 3`.  Obviously, you must create the frontports on the same device(s) that you created the rearports on.

I would suggest you start simple.  Make:
* a CSV to create one rearport on a device
* a CSV to create one corresponding frontport on the same device

When you have that working, then move on to multiple rearports and multiple frontports.
